<a href="https://colab.research.google.com/github/mailnami-coder/python-basics/blob/main/Variables.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Python Basics - Data Types, Variables, and Operators

1. Python Data Types

Numeric Types


*   int (Integer)
*   float (Floating-point)
*   complex (Complex numbers)

Text Type

*   str (String)

Boolean Type

*  bool (True/False)




In [1]:
# Examples of different data types
integer_num = 42
float_num = 3.14
complex_num = 2 + 3j
text = "Hello, Python!"
is_python_fun = True

# Check types
print(f"Type of {integer_num}: {type(integer_num)}")
print(f"Type of {float_num}: {type(float_num)}")
print(f"Type of {complex_num}: {type(complex_num)}")
print(f"Type of {text}: {type(text)}")
print(f"Type of {is_python_fun}: {type(is_python_fun)}")

Type of 42: <class 'int'>
Type of 3.14: <class 'float'>
Type of (2+3j): <class 'complex'>
Type of Hello, Python!: <class 'str'>
Type of True: <class 'bool'>


2. Variables

Rules for Python variables:

*   Must start with a letter or underscore
*   Can contain letters, numbers, and underscores
*   Case-sensitive
*   Cannot use reserved keywords

In [2]:
# Variable assignment examples
name = "Alice"
age = 25
height = 1.75
_private_var = "secret"

# Multiple assignment
x, y, z = 1, 2, 3

print(f"Name: {name}")
print(f"Age: {age}")
print(f"Height: {height}")
print(f"x: {x}, y: {y}, z: {z}")

Name: Alice
Age: 25
Height: 1.75
x: 1, y: 2, z: 3


In [6]:
# Another example of multiple assignment
first_name, last_name, age_group = "John", "Doe", "Adult"
print(f"Name: {first_name} {last_name}, Age Group: {age_group}")

Name: John Doe, Age Group: Adult


3. Operators

Arithmetic Operators

*   `+` (Addition)
*   `-` (Subtraction)
*   `*` (Multiplication)
*   `/` (Division)
*   `//` (Floor Division)
*   `%` (Modulus)
*   `**` (Exponentiation)

In [3]:
# Arithmetic operations
a = 10
b = 3

print(f"Addition: {a + b}")
print(f"Subtraction: {a - b}")
print(f"Multiplication: {a * b}")
print(f"Division: {a / b}")
print(f"Floor Division: {a // b}")
print(f"Modulus: {a % b}")
print(f"Exponentiation: {a ** b}")

Addition: 13
Subtraction: 7
Multiplication: 30
Division: 3.3333333333333335
Floor Division: 3
Modulus: 1
Exponentiation: 1000


The following code demonstrates the difference between standard division (`/`) and floor division (`//`):

*   **Standard Division (`/`)**: Returns a float, including any fractional part.
*   **Floor Division (`//`)**: Returns an integer if both operands are integers, or a float with `.0` if either operand is a float. It truncates the fractional part, effectively rounding down to the nearest whole number.

For example:

*   `10 / 3` results in `3.3333333333333335`.
*   `10 // 3` results in `3`.
*   `10 / 4` results in `2.5`.
*   `10 // 4` results in `2`.

In [7]:
# Demonstrating the difference between division and floor division

num1 = 10
num2 = 3

# Standard division
result_division = num1 / num2
print(f"Standard Division ({num1} / {num2}): {result_division}")

# Floor division
result_floor_division = num1 // num2
print(f"Floor Division ({num1} // {num2}): {result_floor_division}")

num3 = 10
num4 = 4

# Standard division
result_division_2 = num3 / num4
print(f"Standard Division ({num3} / {num4}): {result_division_2}")

# Floor division
result_floor_division_2 = num3 // num4
print(f"Floor Division ({num3} // {num4}): {result_floor_division_2}")


Standard Division (10 / 3): 3.3333333333333335
Floor Division (10 // 3): 3
Standard Division (10 / 4): 2.5
Floor Division (10 // 4): 2


In [4]:
# Exercise 1: Temperature Converter
def celsius_to_fahrenheit(celsius):
    return (celsius * 9/5) + 32

# Test the function
temp_c = 25
temp_f = celsius_to_fahrenheit(temp_c)
print(f"{temp_c}°C is equal to {temp_f}°F")

25°C is equal to 77.0°F


In [5]:
# Exercise 2: Simple Interest Calculator
def calculate_simple_interest(principal, rate, time):
    return (principal * rate * time) / 100

# Test the function
p = 1000  # Principal amount
r = 5     # Rate of interest
t = 2     # Time in years

interest = calculate_simple_interest(p, r, t)
print(f"Simple Interest: ${interest}")

Simple Interest: $100.0


Python is Case sentive.

In [8]:
# Demonstrating case sensitivity

my_variable = "This is the lowercase variable"
My_Variable = "This is the uppercase variable"

print(f"Value of my_variable: {my_variable}")
print(f"Value of My_Variable: {My_Variable}")

# Attempting to print a non-existent variable (e.g., if case was ignored)
# Uncommenting the line below would cause a NameError
# print(my_Variable)

Value of my_variable: This is the lowercase variable
Value of My_Variable: This is the uppercase variable


The `chr()` function takes an ASCII value as input and returns its corresponding character.

*   **Input**: An ASCII integer value.
*   **Output**: The character represented by that ASCII value.

In [13]:
print(chr(65))
print(chr(65+32))


A
a


The `ord()` function is the inverse of `chr()`. It takes a single character as input and returns its corresponding ASCII integer value.

*   **Input**: A single character string.
*   **Output**: The ASCII integer value of the character.

In [14]:
# Demonstrating the ord() function
print(ord('A'))
print(ord('a'))
print(ord('!'))

65
97
33
